### ISSUE TITLE
Not finishing JumpingForward on certain snippet makes Ultisnips behave weird.

### ISSUE THREAD

**Issue by vimeitor** [SirVer/ultisnips#263]
Hello,

TLDR: https://asciinema.org/a/8725  (video of the weird behavior)

Long answer: Let's say I'm in a C++ file and type 'Inc', expand to '#include <.h>', change whatever is inside <>, and immediately go to normal mode (don't finish JumpingForward to the end). If now I type 'vector' in a new line and expand it, I will get something like 'std::vector<char> v'. If I JumpForward, I will go to 'char'; if I JumpForward again, I will go to v; if I do it again, Ultisnips will attemp to sort of jump to the end of #include<.h>, since we didn't finish all the JumpForward.

I have the latest version of Ultisnips, I'm using Houza's snippets, and I have Vim 7.4.

I think the issue is Ultisnips. If for any reason Ultisnips weren't the reason of this behavior, I would be very sorry for making you lose time (I already am :/ ).

Take care, and thank you very much for maintaining Ultisnips. It's been very helpful.

Adria

**Comment by SirVer** [comment]
This is fixed in master. At least I can no longer repro it.
